File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Prefix Sum/1-2270 Number of ways to split array/main.py
from typing import List

class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        n = len(nums)
        prefix = [nums[0]]
        for i in range(1, n):
            prefix.append(prefix[len(prefix) - 1] + nums[i])

        ans = 0
        for i in range(n - 1):
            leftSection = prefix[i]
            rightSection = prefix[n - 1] - prefix[i]
            if leftSection >= rightSection:
                ans += 1

        return ans

File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Prefix Sum/1-2270 Number of ways to split array/main.ts
function waysToSplitArray(nums: number[]): number {
    let n = nums.length;

    let prefix: number[] = [nums[0]];
    for (let i = 1; i < n; i++)
        prefix.push(prefix[prefix.length - 1] + nums[i]);

    let ans = 0;
    for (let i = 0; i < n - 1; i++) {
        let leftSection = prefix[i];
        let rightSection = prefix[n - 1] - prefix[i];
        if (leftSection >= rightSection) ans++;
    }

    return ans;
};